src/Microsoft.VisualStudio.Editor.Razor/ProjectPathProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal abstract class ProjectPathProvider : ILanguageService
+    {
+        public abstract bool TryGetProjectPath(ITextBuffer textBuffer, out string filePath);
```

Comment:
LiveShare needs to ask the Host for the project path. It will eventually use a cached value that's already be computed and stored on the text buffer.